modules/app_gke/variables.tf

```diff
@@ -21,4 +21,15 @@ variable "subnetwork" {
 variable "machine_type" {
   type    = string
   default = "n1-standard-4"
-}
\ No newline at end of file
+}
+
+variable "gke_version" {
+  description = "Default GKE version"
+  type        = string
+  default     = "1.22.12-gke.500"
+
+  validation {
+    condition     = regex("^1.2[2-4].*", var.gke_version)
```

Comment:
This has been tested.

```
for a in "1.22.16-gke.2000" "1.23.14-gke.1800" "1.24.9-gke.1500" "1.21.14-gke.14600" "1.25" ; do echo "length(regexall(\"^1.2[2-4].*\", \"${a}\")) > 0" | terraform console; done
true
true
true
false
false
````

Reply:
This still seems wrong. You are limiting the versions to 1.24, 1.23, 1.24, means we would need to update it on version 1.25

Something like this would make more sense
```
^1.[0-9]*.[0-9]*-gke.[0-9]*$
```

Reply:
The GKE valid versions in stable channel are:

```
  - 1.24.9-gke.1500
  - 1.23.14-gke.1800
  - 1.22.16-gke.2000
  - 1.21.14-gke.14600
  - 1.21.14-gke.14100
```

But I'm fine to open to any other version if you believe this is what we want.